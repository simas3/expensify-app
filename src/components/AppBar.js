import React from 'react'
import AppBar from 'material-ui/AppBar'

const TopBar = (props) =>
    (
        <AppBar
            title="Expensify"
            iconClassNameRight="muidocs-icon-navigation-expand-more"
            onLeftIconButtonClick={props.handleToggle}
        />
    )


export default TopBar


