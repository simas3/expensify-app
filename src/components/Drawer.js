import React from 'react';
import map from 'lodash/map'
import { List } from 'material-ui/List'
import Drawer from 'material-ui/Drawer'
import TopBar from './AppBar'
import links from '../data/linksData'
import SideBarItem from './SideBarItem'

export default class AppBarDrawer extends React.Component {

    constructor(props) {
        super(props);
        this.state = { open: false };
    }


    handleToggle = () => this.setState({ open: !this.state.open });

    handleClose = () => this.setState({ open: false });

    render() {
        return (
            <div>
                <TopBar handleToggle={this.handleToggle} />
                <Drawer
                    docked={false}
                    width={200}
                    open={this.state.open}
                    onRequestChange={(open) => this.setState({ open })}

                >
                    <List>
                        {map(links, ({ isExact, linkTo, text }, key) => {
                            return (
                                <SideBarItem
                                    isExact={isExact}
                                    linkTo={linkTo}
                                    primaryText={text}
                                    onClick={this.handleClose}
                                    key={key}
                                />
                            )
                        })}
                    </List>
                </Drawer>
            </div>
        );
    }
}