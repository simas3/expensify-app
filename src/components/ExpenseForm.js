import React, { Component } from 'react'

import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import DatePicker from 'material-ui/DatePicker';
import moment from 'moment'



export default class ExpenseForm extends Component {

    constructor(props) {
        super(props)

        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount / 100).toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            erroramount: '',
            errordescription: ''
        }
    }


    onDescriptionChange = (e) => {
        const description = e.target.value
        this.setState(() => ({ description, errordescription: '' }))
    }

    onNoteChange = (e) => {
        const note = e.target.value
        this.setState(() => ({ note }))
    }

    onAmountChange = (e) => {
        const amount = e.target.value
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount, erroramount: '' }))
        }

    }

    onDateChange = (event, date) => {
        this.setState(() => ({ createdAt: date }))
    }

    onSubmit = (e) => {
        e.preventDefault()
        if (!this.state.description || !this.state.amount) {
            if (!this.state.description) {
                this.setState(() => ({ errordescription: 'Please Enter Description' }))
            }
            if (!this.state.amount) {
                this.setState(() => ({ erroramount: 'Please Enter Amount' }))

            }
        } else {
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                createdAt: +this.state.createdAt,
                note: this.state.note
            })
        }
    }

    render() {
        return (
            <div>
                Expense Form
                <form onSubmit={this.onSubmit}>
                    <TextField
                        autoFocus
                        errorText={this.state.errordescription}
                        floatingLabelText="Description"
                        onChange={this.onDescriptionChange}
                        value={this.state.description}
                    />
                    <br />
                    <TextField
                        errorText={this.state.erroramount}
                        value={this.state.amount}
                        floatingLabelText="Amount"
                        onChange={this.onAmountChange}
                    />
                    <br />
                    <br />
                    <DatePicker
                        value={this.state.createdAt}
                        autoOk={true}
                        floatingLabelText="Choose date"
                        mode="landscape"
                        onChange={this.onDateChange}
                    />
                    <TextField
                        value={this.state.note}
                        onChange={this.onNoteChange}
                        floatingLabelText="Add a note for your expense"
                        multiLine={true}
                        rows={3}
                    />
                    <br />
                    <br />
                    <RaisedButton
                        type="submit"
                        label="Add Expense"
                        primary={true}
                        onClick={() => {
                        }
                        } />
                </form>
            </div>
        )
    }
}
