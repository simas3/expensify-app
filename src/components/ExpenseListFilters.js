import React, { Component } from 'react'
import TextField from 'material-ui/TextField';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';
import { connect } from 'react-redux'
import { setTextFilter, sortByAmount, sortByDate, setEndDate, setStartDate } from '../actions/filters'
import { DateRangePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

const items = [
    <MenuItem key={1} value="date" primaryText="Date" />,
    <MenuItem key={2} value="amount" primaryText="Amount" />
];


export class ExpenseListFilers extends Component {

    state = {
        calendarFocused: null
    }

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }

    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }))
    }

    onSelectChange = (event, index, value) => {
        if (value === 'amount') {
            this.props.sortByAmount()
        } else if (value === 'date') {
            this.props.sortByDate()
        }
    }

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }

    render() {
        return (
            <div>
                <TextField
                    floatingLabelText="Enter you filter text"
                    onChange={this.onTextChange}
                />
                <br />
                <SelectField
                    value={this.props.filters.sortBy}
                    onChange={this.onSelectChange}
                    floatingLabelText="Select To Sort Buy"
                >
                    {items}
                </SelectField>
                <br />
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    startDateId="your_unique_start_date_id"
                    endDateId="your_unique_end_date_id"
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    showClearDates={true}
                    isOutsideRange={() => false}
                />
            </div>
        )
    }
}


const mapDistpatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
})

const mapStateToProps = (state) => ({
    filters: state.filters
})


export default connect(mapStateToProps, mapDistpatchToProps)(ExpenseListFilers)