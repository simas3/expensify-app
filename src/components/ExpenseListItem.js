import React from 'react'
import { Link } from 'react-router-dom'


export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link style={{ color: '#353b48' }} to={`/edit/${id}`}>
            <h3>Description: {description}</h3>
        </Link>
        <p>Amount: {amount}</p>
        <p>Created: {createdAt}</p>
    </div>
)



export default ExpenseListItem


