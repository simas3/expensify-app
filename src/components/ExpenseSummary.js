import React, { Component } from 'react'
import { connect } from 'react-redux'
import expensesTotalAmount from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'


export class ExpenseSummary extends Component {
    render() {
        return (
            <div>
                {
                    this.props.expensesCount === 1 ?
                        (<p>Viewing 1 expense totalling {numeral(this.props.expensesTotal / 100).format('$0,0.00')}</p>)
                        : this.props.expensesCount > 1 ?
                            (<p>Viewing {this.props.expensesCount} expenses totalling {numeral(this.props.expensesTotal / 100).format('$0,0.00')}</p>)
                            : (<p></p>)
                }
            </div>
        )
    }
}


const mapStateToProps = (state) => ({
    expensesCount: selectExpenses(state.expenses, state.filters).length,
    expensesTotal: expensesTotalAmount(selectExpenses(state.expenses, state.filters))
})


export default connect(mapStateToProps)(ExpenseSummary)