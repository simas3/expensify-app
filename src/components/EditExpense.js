import React, { Component } from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'
import RaisedButton from 'material-ui/RaisedButton';



export class EditExpense extends Component {

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense)
        this.props.history.push('/dashboard')
    }

    onClick = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id })
        this.props.history.push('/dashboard')
    }

    render() {
        return (
            <div className="container">
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit} />
                <br />
                <RaisedButton
                    label="Remove"
                    primary={true}
                    onClick={this.onClick} />
            </div>
        )
    }
}


const mapDispatchToProps = (dispatch, props) => (
    {
        startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
        startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
    }
)

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);