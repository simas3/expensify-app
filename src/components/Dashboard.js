import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'


const Dashboard = () => (
  <div className="container">
    <ExpenseListFilters />
    <ExpenseList />
  </div>
)


export default Dashboard;