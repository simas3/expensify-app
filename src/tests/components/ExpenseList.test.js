import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseList } from '../../components/ExpenseList'
import expenses from '../fixtures/expenses'

test('render list', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />)
    expect(wrapper).toMatchSnapshot()
})


test('render with empty list', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />)
    expect(wrapper).toMatchSnapshot()
})