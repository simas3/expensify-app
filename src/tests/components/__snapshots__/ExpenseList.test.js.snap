// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render list 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    expenses={
        Array [
            Object {
              "amount": 195,
              "createdAt": 0,
              "description": "Coffee",
              "id": "1",
              "note": "",
            },
            Object {
              "amount": 109500,
              "createdAt": -345600000,
              "description": "Rent",
              "id": "2",
              "note": "",
            },
            Object {
              "amount": 4000,
              "createdAt": 345600000,
              "description": "Bat",
              "id": "3",
              "note": "",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          amount={195}
          createdAt={0}
          description="Coffee"
          id="1"
          note=""
/>,
        <Unknown
          amount={109500}
          createdAt={-345600000}
          description="Rent"
          id="2"
          note=""
/>,
        <Unknown
          amount={4000}
          createdAt={345600000}
          description="Bat"
          id="3"
          note=""
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "amount": 195,
          "createdAt": 0,
          "description": "Coffee",
          "id": "1",
          "note": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "amount": 109500,
          "createdAt": -345600000,
          "description": "Rent",
          "id": "2",
          "note": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "amount": 4000,
          "createdAt": 345600000,
          "description": "Bat",
          "id": "3",
          "note": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            amount={195}
            createdAt={0}
            description="Coffee"
            id="1"
            note=""
/>,
          <Unknown
            amount={109500}
            createdAt={-345600000}
            description="Rent"
            id="2"
            note=""
/>,
          <Unknown
            amount={4000}
            createdAt={345600000}
            description="Bat"
            id="3"
            note=""
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "amount": 195,
            "createdAt": 0,
            "description": "Coffee",
            "id": "1",
            "note": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "amount": 109500,
            "createdAt": -345600000,
            "description": "Rent",
            "id": "2",
            "note": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "amount": 4000,
            "createdAt": 345600000,
            "description": "Bat",
            "id": "3",
            "note": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`render with empty list 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    expenses={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        No expenses available
</p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "No expenses available",
      },
      "ref": null,
      "rendered": "No expenses available",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          No expenses available
</p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No expenses available",
        },
        "ref": null,
        "rendered": "No expenses available",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
